crates/miden-objects/src/note/note_tag.rs

```diff
@@ -273,19 +326,19 @@ impl TryFrom<Felt> for NoteTag {
 
 impl From<NoteTag> for u32 {
     fn from(value: NoteTag) -> Self {
-        value.0
+        value.as_u32()
     }
 }
 
 impl From<NoteTag> for u64 {
     fn from(value: NoteTag) -> Self {
-        value.0 as u64
+        value.as_u32() as u64
     }
 }
```

Comment:
For similar reasons, I would also remove this implementation and keep only the `From<NoteTag> for Felt`. We'll have to update a few places where we use it, but that's easily fixed using `Felt::from(note_tag)`.